numpy/lib/npyio.py

```diff
@@ -86,10 +86,12 @@ def __dir__(self):
         return object.__getattribute__(self, '_obj').keys()
 
 
-def zipfile_factory(*args, **kwargs):
+def zipfile_factory(file, mode='r', *args, **kwargs):
```

Comment:
I reverted the type signature, although I did add an `assert`. With the previous version, a 0-length `*args` would cause an error from `zipfile.Zipfile`; this just changes the exception that would be raised from a `TypeError` to an `AssertionError`. Is that ok, or should I also preserve the assertion type?